src/index.ts

```diff
@@ -12,30 +17,76 @@ import { FileReaderGame } from './types';
  * @returns
  */
 export async function main(path: string) {
+  console.time("Total Execution Time");
+
+  console.time("Task 1: FileReader")
   const gamesGenerator = gameChunks(path);
   const games: FileReaderGame[] = [];
+  let gameCounter = 0;
   for await (const game of gamesGenerator) {
+    gameCounter++;
+    if (gameCounter % 20 == 0) {
+      console.log('number of games analyzed: ', gameCounter);
+    }
     games.push(game);
+
+    // const siteLink = game.metadata[1].match(/"(.*?)"/)[1];
+    // console.log(`lichess link to game played: ${siteLink}`);
   }
+  console.timeEnd("Task 1: FileReader")
+
+  console.time("Task 2: getMoveDistanceSetOfGames")
   const {
     pieceThatMovedTheFurthest,
     maxDistance,
     gameCount,
     siteWithFurthestPiece,
     totalDistanceMap,
+    furthestCollectiveDistance,
+    gameLinkWithFurthestCollectiveDistance,
   } = await getMoveDistanceSetOfGames(games);
+  console.timeEnd("Task 2: getMoveDistanceSetOfGames")
 
+  console.time("Task 3: getAverageDistance")
   const { 
     pieceWithHighestAverageDistance,
     maxAverageDistance,
    } = getAverageDistance(totalDistanceMap, gameCount);
+  console.timeEnd("Task 3: getAverageDistance")
 
+  console.time("Task 4: getKillDeathRatios")
   const {
     killDeathRatios,
     killsDeathsAssistsMap,
     pieceWithHighestKillDeathRatio,
   } = await getKillDeathRatios(games);
+  console.timeEnd("Task 4: getKillDeathRatios")
 
+  console.time("Task 5: getGameWithMostMoves")
```

Comment:
logging all of this will slow down things bc of terminal limitations. Let's not do it in the main fn, we can add this in a unit testing framework to check performance

Reply:
but good to track time!